examples/cpp/dds/DiscoveryServerExample/arg_configuration.h

```diff
@@ -0,0 +1,379 @@
+            {
+                return option::ARG_OK;
+            }
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires an <IPaddress> v4 or v6 argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus Transport(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            // we must check if it is a correct ip address plus port number
+            std::string transport = std::string(option.arg);
+            if (
+                    // transport == "shm" ||
+                    transport == "udpv4" ||
+                    transport == "udpv6"
+                )
+            {
+                return option::ARG_OK;
+            }
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires an IPaddress[:portnumber] argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+};
+
+enum  optionIndex
+{
+    UNKNOWN_OPT,
+    HELP,
+
+    TOPIC,
+    SAMPLES,
+    INTERVAL,
+    TRANSPORT,
+
+    CONNECTION_ADDRESS,
+    CONNECTION_PORT,
+    CONNECTION_DISCOVERY_SERVER_ID,
+
+    LISTENING_ADDRESS,
+    LISTENING_PORT,
+    LISTENING_DISCOVERY_SERVER_ID,
+    TIMEOUT,
+};
+
+const option::Descriptor usage[] = {
+    {UNKNOWN_OPT, 0, "", "", Arg::None,
+        "Usage: DiscoveryServerExample <publisher|subscriber|server>\n\nGeneral options:" },
+    {
+        HELP,
+        0,
+        "h",
+        "help",
+        Arg::None,
+        "  -h \t--help  \tProduce help message."
+    },
+
+    /// PUBLISHER OPTIONS
+    {UNKNOWN_OPT, 0, "", "", Arg::None, "\nPublisher options:"},
+    {
+        TOPIC,
+        0,
+        "n",
+        "topic",
+        Arg::String,
+        "  -n <topic_name> \t--topic=<topic_name>  \tTopic name (Default: HelloWorldTopic)."
+    },
+    {
+        SAMPLES,
+        0,
+        "s",
+        "samples",
+        Arg::Numeric,
+        "  -s <num> \t--samples=<num>  \tNumber of samples to send (Default: 0 => infinite samples)."
+    },
+    {
+        INTERVAL,
+        0,
+        "i",
+        "interval",
+        Arg::Numeric,
+        "  -i <num> \t--interval=<num>  \tTime between samples in milliseconds (Default: 100)."
+    },
+    {
+        CONNECTION_ADDRESS,
+        0,
+        "c",
+        "connection-address",
+        Arg::String,
+        "  -c <IPaddress> \t--connection-address=<IPaddress>  \tServer address (Default address: ::1)."
+    },
+    {
+        CONNECTION_PORT,
+        0,
+        "p",
+        "connection-port",
+        Arg::Numeric,
+        "  -p <num> \t--connection-port=<num>  \tServer listening port (Default port: 16166)."
+    },
+    {
+        TRANSPORT,
+        0,
+        "t",
+        "transport",
+        Arg::Transport,
+      "  -t <trans>\t--transport <trans> \tUse Transport Protocol [udpv4|udpv6] (UDPv6 by default)."
```

Comment:
Please, update the example to use UDPv4 by default. This implies:
- Change in main the default arguments
- Change this help messages

Reply:
It must change also de default value (and its messages) for the listening and connection address:
`::1` -> `127.0.0.1`